api/hasura/actions/_handlers/createEpoch.ts

```diff
@@ -108,6 +108,23 @@ async function handler(request: VercelRequest, response: VercelResponse) {
   insertNewEpoch(response, input);
 }
 
+function validateMonthlyInput(
+  input: z.infer<typeof zMonthlyInputSchema>
+): ErrorReturn {
+  const { start_date, end_date } = input;
```

Comment:
Why does end_date need to be input in this case? Shouldn't we just calc it?

Reply:
I saw the comment in the PR and i think i understand, but i'm worried about user confusion, maybe we need to discuss . Does the user have to specify a week number?

Reply:
@CryptoGraffe a date is picked from a datepicker and the UI informs them of the pattern like "The epoch repeats on the **2nd Tuesday** of each month"